src/controllers/confirm.company.controller.ts

```diff
@@ -13,18 +12,18 @@ import logger from "../utils/logger";
  */
 
 export const route = (req: Request, res: Response, next: NextFunction) => {
-    if (req.session && req.session.data) {
-        const company: ObjectionCompanyProfile = getValueFromObjectionsSession(req.session, SESSION_COMPANY_PROFILE);
-        return res.render(Templates.CONFIRM_COMPANY, {
-            company,
-            templateName: Templates.CONFIRM_COMPANY,
-        });
-    } else {
-        logger.errorRequest(req, "Confirm Company - no company data found in session");
-        // TODO - implement generic error page to be called with next(ERROR())
-        // same as PTF eg next(e)
-        return next();
+  if (req.session) {
+    try {
+      const company: ObjectionCompanyProfile = retrieveCompanyProfileFromObjectionsSession(req.session);
+      return res.render(Templates.CONFIRM_COMPANY, {
+        company,
+        templateName: Templates.CONFIRM_COMPANY,
+      });
+    } catch (e) {
+      logger.errorRequest(req, "Error retrieving company profile from session");
+      return next(e);
     }
+  }
```

Comment:
I would either add a next(new Error("no session")) as an else block, or remove the if(req.session) from the top and let it fail if no session.